packages/bpk-component-carousel/src/utils.tsx

```diff
@@ -25,15 +25,27 @@ export function useScrollToInitialImage(
   initialImageIndex: number,
   imagesRef: MutableRefObject<Array<HTMLElement | null>>,
 ) {
-  useEffect(() => {
-    const element = imagesRef.current[initialImageIndex];
+  const handleIntersecting = (index: number) => {
+    const element = imagesRef.current[index];
     if (element) {
       element.scrollIntoView({
```

Comment:
`element.scrollIntoView` can cause reflow performance issues, even if the image is not in the viewport. Therefore, it is necessary to use IntersectionObserver to monitor the element.